### PR TITLE
DEVDOCS-4670: [new] Carts V3, add cart metafield endpoints

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-4670]

## What changed?
Added cart metafield endpoints

## Anything else?
Related PRs, salient notes, etc

[DEVDOCS-4670]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4670?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by becomevocal — CHANGES_REQUESTED** [review]
Two things that would make this ready-to-go:
- `cart_id` needs to be listed as a string type. It's currently listed as an integer a few times in this PR. It's a string because they reference the UUID of the cart.
-  "The ID of the `Metafield` to which the transactions belong" as the description of `metafield_id` doesn't sound completely accurate? Because transactions aren't intimately attached to metafields. Simplifying this to "The unique ID of the `Metafield`" would be more accurate.

File: reference/carts.v3.yml [hunk 1]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+      description: |
+        Common Metafield properties.
+      x-internal: false
+      properties:
+        permission_set:
+          type: string
+          description: |
+            Determines the visibility and writeability of the field by other API consumers.
+
+            |Value|Description
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3331]:
```suggestion
            | Value | Description |
```

File: reference/carts.v3.yml [hunk 2]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |-|-|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3332]:
```suggestion
            | :--- | :--- |
```
I think we've been doing this for our tables

File: reference/carts.v3.yml [hunk 3]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`app_only`|Private to the app that owns the field|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3333]:
```suggestion
            | `app_only` | Private to the app that owns the field. |
```

File: reference/carts.v3.yml [hunk 4]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`read`|Visible to other API consumers|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3334]:
```suggestion
            | `read` | Visible to other API consumers. |
```

File: reference/carts.v3.yml [hunk 5]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`write`|Open for reading and writing by other API consumers|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3335]:
```suggestion
            | `write` | Open for reading and writing by other API consumers. |
```

File: reference/carts.v3.yml [hunk 6]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`read_and_sf_access`|Visible to other API consumers, including on storefront|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3336]:
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```

File: reference/carts.v3.yml [hunk 7]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+      description: |
+        Common Metafield properties.
+      x-internal: false
+      properties:
+        permission_set:
+          type: string
+          description: |
+            Determines the visibility and writeability of the field by other API consumers.
+
+            |Value|Description
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3253]:
```suggestion
            | Value | Description |
```

File: reference/carts.v3.yml [hunk 8]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |-|-|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3254]:
```suggestion
            | :--- | :--- |
```
I believe we've been doing this for our tables

File: reference/carts.v3.yml [hunk 9]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`app_only`|Private to the app that owns the field|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3255]:
```suggestion
            | `app_only` | Private to the app that owns the field. |
```

File: reference/carts.v3.yml [hunk 10]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`read`|Visible to other API consumers|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3256]:
```suggestion
            | `read` | Visible to other API consumers. |
```

File: reference/carts.v3.yml [hunk 11]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`write`|Open for reading and writing by other API consumers|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3257]:
```suggestion
            | `write` | Open for reading and writing by other API consumers. |
```

File: reference/carts.v3.yml [hunk 12]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`read_and_sf_access`|Visible to other API consumers, including on storefront|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3258]:
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```

File: reference/carts.v3.yml [hunk 13]
```diff
@@ -2969,6 +3186,313 @@ components:
...
+            |`write_and_sf_access`|Open for reading and writing by other API consumers, including on storefront|
```
Comment (bc-andreadao) [inline reference/carts.v3.yml:3259]:
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```

**Review by bc-andreadao — COMMENTED** [review]
added table markdown suggestions

File: reference/carts.v3.yml [hunk 14]
```diff
@@ -1,4 +1,4 @@
-openapi: '3.0.3'
+openapi: 3.0.3
```
Comment (slsriehl) [inline reference/carts.v3.yml:1]:
```suggestion
openapi: '3.0.3'
```

string type per oas spec

**Review by slsriehl — APPROVED** [review]
this was a lot of work. well done!

**Pull request merged**